be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/validators/CoinValidator.scala

```diff
@@ -24,7 +24,10 @@ case object CoinValidator extends MessageDataContentValidator {
         } else if (!data.transaction.checkSignatures()) {
           Right(validationError("bad signature"))
         } else {
-          Left(rpcMessage)
+          data.transaction.sumOutputs() match {
+            case Left(err) => Right(validationError(err.getMessage()))
+            case Right(_) => Left(rpcMessage)
```

Comment:
Hmmm maybe I am missing something here but shouldn't Left and Right be interchanged here?